src/lex/cpp.rs

```diff
@@ -1832,4 +1857,51 @@ h",
         assert!(cpp("#define f(x) #y\nf(0)").any(|x| x.is_err()));
         assert!(cpp("#define f(x) #+\nf(0)").any(|x| x.is_err()));
     }
+
+    #[test]
+    fn builtins_line() {
+        assert_same("__LINE__", "1");
+        assert_same("\n\n\n\n\n\n\n\n\n__LINE__", "10");
+        assert_same(
+            "#ifdef __LINE__
+            1
+            #endif",
+            "1",
+        );
+        assert_same(
+            "
+            
+            #if __LINE__ == 3
+            1
+            #endif",
+            "1",
+        );
```

Comment:
Can you add a test case where LINE is used twice to make sure the old definition gets overridden?

Reply:
Done